client/client.go

```diff
@@ -2572,8 +2572,9 @@ func (c *Client) deriveToken(alloc *structs.Allocation, taskNames []string, vcli
 		AllocID:  alloc.ID,
 		Tasks:    verifiedTasks,
 		QueryOptions: structs.QueryOptions{
-			Region:     c.Region(),
-			AllowStale: false,
+			Region:        c.Region(),
+			AllowStale:    false,
+			MinQueryIndex: alloc.ModifyIndex,
```

Comment:
Hm, should we just use `alloc.CreateIndex` here? It shouldn't matter, but CreateIndex represents the _earliest_ index at which this alloc exists so it might return sooner?